hivemind/moe/server/dht_handler.py

```diff
@@ -16,21 +16,24 @@
     split_uid,
 )
 from hivemind.p2p import PeerID
-from hivemind.utils import MPFuture, get_dht_time
+from hivemind.utils import MAX_DHT_TIME_DISCREPANCY_SECONDS, MPFuture, get_dht_time
 
 
 class DHTHandlerThread(threading.Thread):
-    def __init__(self, experts, dht: DHT, update_period: int = 5, **kwargs):
+    def __init__(self, experts, dht: DHT, update_period: int = 30, expiration: Optional[int] = None, **kwargs):
         super().__init__(**kwargs)
+        if expiration is None:
+            expiration = max(2 * update_period, MAX_DHT_TIME_DISCREPANCY_SECONDS)
```

Comment:
Even though I can understand why "the maximum time discepancy between DHT instances" kinda makes sense for a lower bound on the record expiration time, it's still not quite obvious why we use it here and not in all other cases for the expiration time limits. Maybe it's best to designate another module-local constant for this, since its purpose differs from measuring the discrepancy between two timestamps?

Reply:
I respectfully disagree: the max discrepancy is introduced precisely in order to account for the allowed discrepancy between nodes.